src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7519,6 +7519,8 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                             property.OriginalDefinition is { ParameterCount: 1 } original &&
                             isIntNotByRef(original.Parameters[0]))
                         {
+                            CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
+                            // note: implicit copy check on the indexer accessor happens in CheckPropertyValueKind
```

Comment:
@RikkiGibson I'm integrating this change into the list-patterns branch, but didn't understand this comment. 
Isn't the line above doing the "implicitty copy check on the indexer accessor"?